File: 정수빈/LeetCode/samsung/binary-search.py
# https://leetcode.com/problems/binary-search/?envType=company&envId=samsung&favoriteSlug=samsung-all
# 난이도: easy
# 해결 x
# 설명: 바이너리 서치, 엄청 쉬운 건데 rp, lp 세팅을 제대로 못함..

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lp = 0
        rp = len(nums) -1
        while lp <= rp:
            mid = (lp + rp) // 2
            middle = nums[mid]
            if middle > target:
                rp = mid - 1
            elif middle == target:
                return mid
            else:
                lp = mid + 1
        return -1
            

File: 장세훈/LeetCode/easy/BinarySearch.java
class BinarySearch {
    /*
            https://leetcode.com/problems/binary-search/submissions/1476054164/
            이진 검색
     */
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length - 1;

        while (left < right) {
            int mid = (left + right) / 2;

            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                right = mid - 1;
            else
                left = mid + 1;
        }

        return -1;
    }
}